services/src/handlers/ebv.rs

```diff
@@ -0,0 +1,671 @@
+//! GEO BON EBV Portal catalog lookup service
+//!
+//! Connects to <https://portal.geobon.org/api/v1/>.
+
+use crate::contexts::Context;
+use crate::error::Result;
+use actix_web::{
+    web::{self, ServiceConfig},
+    FromRequest, Responder,
+};
+use serde::Serialize;
+
+pub(crate) fn init_ebv_routes<C>(base_url: Option<String>) -> Box<dyn FnOnce(&mut ServiceConfig)>
+where
+    C: Context,
+    C::Session: FromRequest,
+{
+    Box::new(move |cfg: &mut web::ServiceConfig| {
+        cfg.app_data(web::Data::new(BaseUrl(
+            base_url.unwrap_or_else(|| BASE_URL.to_owned()),
+        )))
+        .service(web::resource("/classes").route(web::get().to(get_classes::<C>)))
+        .service(web::resource("/datasets/{ebv_name}").route(web::get().to(get_ebv_datasets::<C>)))
+        .service(web::resource("/dataset/{id}").route(web::get().to(get_ebv_dataset::<C>)));
+    })
+}
+
+const BASE_URL: &str = "https://portal.geobon.org/api/v1";
+
+struct BaseUrl(String);
+
+impl AsRef<str> for BaseUrl {
+    fn as_ref(&self) -> &str {
+        &self.0
+    }
+}
+
+mod portal_responses {
+    use serde::Deserialize;
+
+    #[derive(Debug, Deserialize)]
+    pub struct EbvClassesResponse {
+        pub data: Vec<EbvClassesResponseClass>,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct EbvClassesResponseClass {
+        pub ebv_class: String,
+        pub ebv_name: Vec<String>,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct EbvDatasetsResponse {
+        pub data: Vec<EbvDatasetsResponseData>,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct EbvDatasetsResponseData {
+        pub id: String,
+        pub title: String,
+        pub summary: String,
+        pub creator: EbvDatasetsResponseCreator,
+        pub license: String,
+        pub dataset: EbvDatasetsResponseDataset,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct EbvDatasetsResponseCreator {
+        pub creator_name: String,
+        pub creator_institution: String,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct EbvDatasetsResponseDataset {
+        pub pathname: String,
+    }
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct EbvClass {
+    name: String,
+    ebv_names: Vec<String>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct EbvClasses {
+    classes: Vec<EbvClass>,
+}
+
+async fn get_classes<C: Context>(
+    _params: web::Query<()>,
+    base_url: web::Data<BaseUrl>,
+    _session: C::Session,
+    _ctx: web::Data<C>,
+) -> Result<impl Responder> {
+    let base_url = base_url.get_ref().as_ref();
+    let url = format!("{base_url}/ebv");
+
+    dbg!(&url);
+
+    let response = reqwest::get(url)
```

Comment:
maybe log the request (and response code?) s.t. we have an easier time debugging. The same for the other requests